pythran/optimizations/inlining.py

```diff
@@ -0,0 +1,84 @@
+""" Inlining inline functions body. """
+
+from pythran.analyses import Identifiers
+from pythran.passmanager import Transformation
+
+import ast
+import copy
+
+
+class Inlining(Transformation):
+
+    """
+    Inline one line functions.
+
+    >>> import ast
+    >>> from pythran import passmanager, backend
+    >>> pm = passmanager.PassManager("test")
+    >>> node = ast.parse('''
+    ... def foo(a):
+    ...     return 2 * a
+    ... def bar(b):
+    ...     return foo(b) * foo(b)")
+    >>> _, node = pm.apply(Inlining, node)
+    >>> print pm.dump(backend.Python, node)
+    def foo(a):
+        return 2 * a
+    def bar(b):
+        return (2 * b) * (2 * b)
+    """
+
+    def __init__(self):
+        """ fun : Function {name :body} for inlinable functions. """
+        self.fun = dict()
+        self.update = False
+        super(Inlining, self).__init__()
+
+    def visit_Module(self, node):
+        """
+        Gather inlinable function.
+
+        A function can be inline if:
+            * It is a one line function
+            * It is not recursive
+        """
+        for stmt in node.body:
+            if isinstance(stmt, ast.FunctionDef):
+                if len(stmt.body) == 1:
+                    ids = self.passmanager.gather(Identifiers, stmt.body[0],
+                                                  self.ctx)
+                    if stmt.name not in ids:
+                        self.fun[stmt.name] = stmt
+        return self.generic_visit(node)
+
+    def visit_Call(self, node):
+        """ Replace function call by inlined function's body. """
+        if isinstance(node.func, ast.Name) and node.func.id in self.fun:
```

Comment:
no no no, you should rely on aliasing, and not name id, to perform inlining. remember an identifier name has nothing to do with is actual value :-)